### DISCUSSION TITLE
Using VuePress inside existing Vue app

### DISCUSSION THREAD

**Discussion by gvnt** [vuepress/core#102] [Q&A] [answered]
Hello, 

I'm looking for a way to integrate VuePress inside a Vue app. So users would be able to access the documentation from Vue app sidebar menu (with SPA) only if they are connected to the Vue app. 

Can I achieve this without building Vuepress separately ?
Do you have any idea how i can do this ?

**Comment by meteorlxy** [comment, marked as answer]
I would say NO for the complexity.

However, I think it's technically possible:

1. Use VuePress Node API to `app.init()` and `app.prepare()`, then skip the built-in dev / build, and use your own build process. See `@vuepress/cli` package for example usage.
2. Configure your bundler correctly, especially the `alias` and `define` hook that provided by Plugin API. See `@vuepress/bundler-webpack` or `@vuepress/bundler-vite` package for details.
3. Then you need to load the `@vuepress/client` pacakge in your Vue app:
  - If you want to use VuePress as an individual Vue app, consider using the `createVueApp` method that provided by `@vuepress/client/src/app.ts` to create a new Vue app instance and a new Vue Router instance. But I'm not sure if it can work properly with you current Vue app.
  - If you want to integrete VuePress pages into your current Vue app, you need to implement yourself, following the `@vuepress/client` package.

That's interesting, so I provide some ideas for you. It requires you to fully understand the implementation principle of vuepress.

**Reply by gvnt** [reply]
Thank you very much for taking the time to give such a complete answer, I will look at this in more detail 👍
